c7n/policy.py

```diff
@@ -845,6 +845,11 @@ def get_variables(self, variables=None):
         if not variables:
             variables = {}
 
+        if 'mode' in self.data:
+            if 'role' in self.data['mode']:
```

Comment:
we need a better check here, this would need to support both forms, ie. a user passing in an arn as well as passing a role name. as written it would break on existing usage with an arn.

Reply:
I thought about this breaking change and I will update the PR so that it's not breaking for current users